library/bigip_ssl_certificate.py

```diff
@@ -0,0 +1,449 @@
+                             choices=['pem']),
+            cert_pem_file=dict(type='str'),
+            key_pem_file=dict(type='str'),
+            pkcs12_file=dict(type='str'),
+            pkcs12_password=dict(type='str', no_log=True),
+            validate_cert=dict(type='bool', default=True),
+            # overwrite=dict(type='bool', default=False)
+        )
+    )
+    result = {'changed': False}
+    if module.params['connection'] == 'icontrol':
+        if not bigsuds_found:
+            module.fail_json(msg='connection type icontrol requires bigsuds')
+        try:
+            api = BigIPiControl(module.params['server'],
+                                module.params['user'],
+                                module.params['password'],
+                                module.params['validate_cert'])
+        except F5ModuleError as e:
+            module.fail_json(msg='Connection Error: %s' % e)
+    elif module.params['connection'] == 'rest':
+        if not f5sdk_found:
+            module.fail_json(msg='connection type rest requires '
+                             'f5-common-python')
+        api = BigIPRest(module.params['server'],
+                        module.params['user'],
+                        module.params['password'],
+                        module.params['validate_cert'])
+    else:
+        module.fail_json(msg='connection type %s is unsupported' %
+                         module.params['connection'])
+
+    if module.params['state'] == 'present':
+        if module.params['cert_format'] == 'pem':
+            if not (module.params['key_pem_file'] and
+                    module.params['cert_pem_file']):
+                module.fail_json(msg='pem format requires cert_pem_file '
+                                 'and key_pem_file')
+            certargs = {}
+            if not api.exists_cert('%s.crt' % module.params['name']):
+                certargs['certfilename'] = module.params['cert_pem_file']
+            if not api.exists_key('%s.key' % module.params['name']):
+                certargs['keyfilename'] = module.params['key_pem_file']
+            if certargs:
+                certargs['name'] = module.params['name']
+                certargs['cert_format'] = 'pem'
+        if certargs:
+            try:
+                r = api.import_cert(**certargs)
+            except F5ModuleError as e:
+                module.fail_json(msg="F5ModuleError: %s" % e)
+            else:
+                if r:
+                    result['changed'] = True
+    elif module.params['state'] == 'absent':
+        # TODO Delete cert and key separately???
+        try:
+            r = api.delete(module.params['name'])
+        except F5ModuleError as e:
+            module.fail_json(msg="F5ModuleError: %s" % e)
+        except ConnectioError as e:
+            module.fail_json(msg="F5ModuleError: %s" % e)
+        else:
+            if r is True:
+                result['changed'] = True
+
+    module.exit_json(**result)
+
```

Comment:
You'll want to include the F5 module utils here.

`from ansible.module_utils.f5 import *`